ert/storage/_storage.py

```diff
@@ -135,46 +135,93 @@ async def _post_to_server_async(
     return resp
 
 
-async def add_record(url: str, record: Any) -> None:
+def _get_content_type(record_type: ert.data.RecordType) -> str:
+    content_type = (
+        ContentTypeEnum.OCTET_STREAM
+        if record_type == ert.data.RecordType.BYTES
+        else ContentTypeEnum.CSV
+    )
+    return content_type.value
+
+
+def _set_content_header(
+    header: str, record_type: ert.data.RecordType, headers: Optional[DictStrAny] = None
+) -> DictStrAny:
+    content_type = _get_content_type(record_type=record_type)
+    if headers is None:
+        return {header: content_type}
+    headers[header] = content_type
+    return headers
+
+
+async def add_record(url: str, record: ert.data.Record) -> None:
     headers = {
         "Token": StorageInfo.token(),
     }
 
+    assert record.record_type
     if record.record_type != ert.data.RecordType.BYTES:
-        headers["content-type"] = "text/csv"
+        _set_content_header(
+            header="content-type", record_type=record.record_type, headers=headers
+        )
         data = pd.DataFrame([record.data]).to_csv().encode()
         await _post_to_server_async(url=url, headers=headers, data=data)
     else:
         data = {"file": io.BytesIO(record.data)}
         await _post_to_server_async(url=url, headers=headers, files=data)
 
 
-async def load_record(url: str, record_type: Any) -> ert.data.Record:
+def _interpret_series(row: pd.Series, record_type: ert.data.RecordType) -> Any:
+    if record_type not in {item.value for item in ert.data.RecordType}:
+        raise ValueError(
+            f"Unexpected record type when loading numerical record: {record_type}"
+        )
+
+    if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
+        return {int(k): v for k, v in row.to_dict().items()}
+    return (
+        row.to_list()
+        if record_type == ert.data.RecordType.LIST_FLOAT
+        else row.to_dict()
+    )
+
+
+def _response_to_record_collection(
+    response: requests.Response, metadata: _NumericalMetaData
+) -> ert.data.RecordCollection:
+    content_type, *_ = response.headers["Content-Type"].split(";")
+    if content_type not in {item.value for item in ContentTypeEnum}:
+        raise ValueError(f"Response content type {content_type} cannot be processed")
+
+    records: Iterable[ert.data.Record] = (
+        (
+            ert.data.BlobRecord(data=response.content)
+            for _ in range(metadata.ensemble_size)
+        )
+        if content_type == ContentTypeEnum.OCTET_STREAM.value
+        else (
+            ert.data.NumericalRecord(
+                data=_interpret_series(row=row, record_type=metadata.record_type)
+            )
+            for _, row in read_csv(io.BytesIO(response.content)).iterrows()
+        )
+    )
```

Comment:
I guess this is a matter of preference but I personally find very confusing these inline ifs, in the case where they cannot be written in a single line.

Reply:
Remove ternary if block in order to improve readability